drivers/flash/flash_spi.c

```diff
@@ -0,0 +1,382 @@
+          case FLASH_CMD_PP:
+              response = spi_transmit(data, len);
+              break;
+          default:
+              response = FLASH_STATUS_ERROR;
+              break;
+        }
+    }
+
+    spi_stop();
+
+    return response;
+}
+
+void flash_init(void) {
+    spi_init();
+}
+
+flash_status_t flash_erase_chip(void) {
+    flash_status_t response = FLASH_STATUS_SUCCESS;
+
+    /* Wait for the write-in-progress bit to be cleared. */
+    response= spi_flash_wait_while_busy();
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to check WIP flag! [spi flash erase chip]\n");
+        return response;
+    }
+
+    /* Enable writes. */
+    response = spi_flash_write_enable();
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to write-enable! [spi flash erase chip]\n");
+        return response;
+    }
+
+    /* Erase Chip. */
+    bool res = spi_flash_start();
+    if (!res) {
+        dprint("Failed to start SPI! [spi flash erase chip]\n");
+        return FLASH_STATUS_ERROR;
+    }
+    spi_write(FLASH_CMD_CE);
+    spi_stop();
+
+    /* Wait for the write-in-progress bit to be cleared.*/
+    response= spi_flash_wait_while_busy();
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to check WIP flag! [spi flash erase chip]\n");
+        return response;
+    }
+
+    return response;
+}
+
+flash_status_t flash_erase_sector(const void *addr) {
+    flash_status_t response    = FLASH_STATUS_SUCCESS;
+    uintptr_t      target_addr = (uintptr_t)addr;
+
+    /* Check that the address exceeds the limit. */
+    if ((target_addr + (EXTERNAL_FLASH_SECTOR_SIZE << 10)) >= (EXTERNAL_FLASH_SIZE << 10) ||
+        ((target_addr % (EXTERNAL_FLASH_SECTOR_SIZE << 10)) != 0)) {
+        dprintf("Flash erase sector address over limit! [addr:0x%x]\n", (uint32_t)target_addr);
+        return FLASH_STATUS_ERROR;
+    }
+
+    /* Wait for the write-in-progress bit to be cleared. */
+    response= spi_flash_wait_while_busy();
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to check WIP flag! [spi flash erase sector]\n");
+        return response;
+    }
+
+    /* Enable writes. */
+    response = spi_flash_write_enable();
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to write-enable! [spi flash erase sector]\n");
+        return response;
+    }
+
+    /* Erase Sector. */
+    response = spi_flash_transmit((uint8_t)FLASH_CMD_SE, target_addr, NULL, 0);
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to erase sector! [spi flash erase sector]\n");
+        return response;
+    }
+
+    /* Wait for the write-in-progress bit to be cleared.*/
+    response= spi_flash_wait_while_busy();
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to check WIP flag! [spi flash erase sector]\n");
+        return response;
+    }
+
+    return response;
+}
+
+flash_status_t flash_erase_block(const void *addr) {
```

Comment:
```suggestion
flash_status_t flash_erase_block(uint32_t addr) {
```
As per previous, swap over usages of `target_addr`.